push_notifications/models.py

```diff
@@ -90,11 +109,17 @@ class Meta:
 	def send_message(self, message, **kwargs):
 		from .apns import apns_send_message
 
-		return apns_send_message(registration_id=self.registration_id, alert=message, **kwargs)
+		return apns_send_message(registration_id=self.registration_id, alert=message, application_id=self.application_id, **kwargs)
 
 
 # This is an APNS-only function right now, but maybe GCM will implement it
 # in the future.  But the definition of 'expired' may not be the same. Whatevs
-def get_expired_tokens():
+def get_expired_tokens(application_id):
 	from .apns import apns_fetch_inactive_ids
-	return apns_fetch_inactive_ids()
+	return apns_fetch_inactive_ids(application_id)
+
+
+class ApplicationModel(models.Model):
+	application_id = models.CharField(max_length=64, verbose_name=_("Application ID"), unique=True)
+	gcm_api_key = models.TextField(verbose_name=_("GCM API Key"), null=True, blank=True)
+	apns_certificate = models.FileField(verbose_name=_("APNS Certificate"), null=True, blank=True, upload_to='apns_certificates')
```

Comment:
Really it has been added just to test working with FileField-like fields, as well as a sample of the application model which might be much more complicated than this.

As for the text field, the biggest problem is that the following call wants a cettificate file name, not a certificate file content in any form:

```python
sock = ssl.wrap_socket(sock, ssl_version=ssl.PROTOCOL_TLSv1, certfile=certfile, ca_certs=ca_certs)
```

https://github.com/jleclanche/django-push-notifications/pull/260/files#diff-1261e053a28d197d3c73287c5102c54aL51